### PR TITLE
ref: Use wildcard re-exports where applicable

### PR DESCRIPTION
**Pull request by jan-auer**
In some cases, we export the entire public API of a module via the containing
parent module. The internal module structure is supposed to be an implementation
detail for code organization but does not matter to the public API. An example
for this is the Event protocol.

In most cases, we use wildcard exports `pub use module::*` for this. The
`relay-general` crate did not do this. With modern tooling and code search, it
is now equally easy to locate types. The benefit of using a wildcard export is
that we cannot forget to re-export types that would be then missing from the
public signature.

#skip-changelog

### CODE REVIEW

File: relay-metrics/src/protocol.rs [hunk 1]
```diff
@@ -5,11 +5,10 @@ use std::iter::FusedIterator;
 use hash32::{FnvHasher, Hasher};
 use serde::{Deserialize, Serialize};
 
-pub use relay_common::UnixTimestamp;
-
 #[doc(inline)]
 pub use relay_common::{
     CustomUnit, DurationUnit, FractionUnit, InformationUnit, MetricUnit, ParseMetricUnitError,
+    UnixTimestamp,
```
Comment (jjbayer) [inline relay-metrics/src/protocol.rs:11]: Was there a reason this was not under `#[doc(inline)]` before?

**Review by jjbayer — APPROVED** [review]
LGTM!

**Review by olksdr — APPROVED** [review]
:shipit:

File: relay-metrics/src/protocol.rs (hunk 1 above)
Comment (jan-auer) [reply relay-metrics/src/protocol.rs:11]: I don't think so, looks like an oversight.

**Pull request merged**